bug_actiongroup.php

```diff
@@ -204,14 +205,24 @@
 			if( access_has_bug_level( access_get_status_threshold( $f_status, $t_bug->project_id ), $t_bug_id ) ) {
 				if( true == bug_check_workflow( $t_status, $f_status ) ) {
 					# @todo we need to issue a helper_call_custom_function( 'issue_update_validate', array( $t_bug_id, $t_bug_data, $f_bugnote_text ) );
-					bug_set_field( $t_bug_id, 'status', $f_status );
+					$t_bugdata = bug_get( $t_bug_id );
+					$t_bugdata->status = $f_status;
 
 					# Add bugnote if supplied
 					if( !is_blank( $f_bug_notetext ) ) {
-						$t_bugnote_id = bugnote_add( $t_bug_id, $f_bug_notetext, null, $f_bug_noteprivate );
-						bugnote_process_mentions( $t_bug_id, $t_bugnote_id, $f_bug_notetext );
-						# No need to call email_generic(), bugnote_add() does it
-					} else {
+						$add_bugnote_func = function( array $p_bugnote_params ) {
+							$t_bugnote_id = call_user_func_array( 'bugnote_add', $p_bugnote_params );
+							bugnote_process_mentions( $p_bugnote_params[0], $t_bugnote_id, $p_bugnote_params[1] );
+						};
+						$t_bugnote_params = array( $t_bug_id, $f_bug_notetext, null, $f_bug_noteprivate );
+						$t_bugdata->add_update_callback( $add_bugnote_func, array( $t_bugnote_params ) );
+						}
+
+					$t_bugdata->update( false, true );
```

Comment:
calls with parameters ( false, true ) is not readable.  Consider either adding /* paramName */ before each parameter, or create a method that calls this update internally but has a name that reflects the semantics of ( false, true ).